src/schemes/fluid/entropically_damped_sph/system.jl

```diff
@@ -44,12 +44,14 @@ See [Entropically Damped Artificial Compressibility for SPH](@ref edac) for more
                     gravity-like source terms.
 """
 struct EntropicallyDampedSPHSystem{NDIMS, ELTYPE <: Real, IC, M, DC, K, V, TV,
-                                   PF, ST, B, C} <: FluidSystem{NDIMS, IC}
+                                   PF, ST, SRFT, SRFN, B, C} <: FluidSystem{NDIMS, IC}
     initial_condition                 :: IC
     mass                              :: M # Vector{ELTYPE}: [particle]
     density_calculator                :: DC
     smoothing_kernel                  :: K
     smoothing_length                  :: ELTYPE
+    number_density                    :: Int64
+    color                             :: Int64
```

Comment:
Why do these need to be here? They are only used when using surface tension, no? Then they would be better off in the cache.

Reply:
They are currently only needed when a surface normal method is used.

Reply:
The problem is though that these values are needed for all systems but not all systems have a surface normal or surface tension model activated.